packages/legacy/src/index.js

```diff
@@ -0,0 +1,49 @@
+import { resolve } from 'path';
+
+const validName = /^[a-zA-Z_$][a-zA-Z$_0-9]*$/;
+
+export default function legacy(options) {
+  const exports = {};
+  Object.keys(options).forEach((file) => {
+    exports[resolve(file)] = options[file];
+  });
+
+  return {
+    name: 'legacy',
+
+    transform(content, id) {
+      if (id in exports) {
```

Comment:
use return early for reducing nesting instead?

```suggestion
if (!(id in exports) {
  return null;
}
```